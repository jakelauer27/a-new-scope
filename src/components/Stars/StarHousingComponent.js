import React, { Component } from 'react'
import Stars from '../Stars/Stars'

export default class StarHousingComponent extends Component {
  constructor() {
    super()
    this.state = {
      deltaX: 0,
      direction: true
    }
  }

  componentDidMount() {
    requestAnimationFrame(this.tick)
  }

  tick = () => {
    let newX
    this.state.direction ? 
      newX = this.state.deltaX + 0.04
        : newX = this.state.deltaX - 0.04
    if (newX > 180 || newX < 0) {
      this.setState({
        deltaX: newX,
        direction: !this.state.direction
      })
    } else {
      this.setState({
        deltaX: newX
      })
    }
    requestAnimationFrame(this.tick)
  }

  render() {
    return (
      <div className="star-housing">
        <Stars deltaX={this.state.deltaX}/>
      </div>
    )
  }
}