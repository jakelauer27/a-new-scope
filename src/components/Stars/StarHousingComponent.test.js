import React from 'react';
import StarHousingComponent from './StarHousingComponent';
import { shallow } from 'enzyme'

describe('Star Housing Component', () => {
  const wrapper = shallow(<StarHousingComponent />)

  it('should increment deltaX on tick', () => {
    wrapper.instance().tick()
    expect(wrapper.instance().state.deltaX).toEqual(0.025)
  })

  it('should change direction when deltaX reaches 180', () => {
    expect(wrapper.instance().state.direction).toEqual(true)

    wrapper.instance().setState({ deltaX: 180 })
    wrapper.instance().tick()

    expect(wrapper.instance().state.direction).toEqual(false)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})