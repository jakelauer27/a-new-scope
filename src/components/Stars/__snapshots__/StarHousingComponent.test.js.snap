// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star Housing Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StarHousingComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Stars
        deltaX={180.025}
      />,
      "className": "star-housing",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "deltaX": 180.025,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Stars
          deltaX={180.025}
        />,
        "className": "star-housing",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "deltaX": 180.025,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
